buildtest/tools/config.py

```diff
@@ -140,9 +140,43 @@ def check_configuration():
         sys.exit(1)
 
 
+def load_configuration(config_path=None):
+    """load the default configuration file.
+    """
+    init()
+
+    config_path = config_path or BUILDTEST_CONFIG_FILE
+
+    # load the configuration file
+    with open(BUILDTEST_CONFIG_FILE, "r") as fd:
+        config_opts = yaml.safe_load(fd)
+
+    config_opts["BUILDTEST_VERSION"] = BUILDTEST_VERSION
+
+    # if BUILDTEST_MODULEPATH is empty list then check if MODULEPATH is defined
+    # and set result to BUILDTEST_MODULEPATH
+    if not config_opts["BUILDTEST_MODULEPATH"]:
+        if not os.getenv("MODULEPATH"):
+            config_opts["BUILDTEST_MODULEPATH"] = []
+        else:
+
+            # otherwise set this to MODULEPATH
+            tree_list = []
+
+            # check each directory in MODULEPATH and add it to BUILDTEST_MODULEPATH
+            for tree in os.getenv("MODULEPATH", "").split(":"):
+                if os.path.isdir(tree):
+                    tree_list.append(tree)
+                else:
+                    print(f"Skipping module tree {tree} because path does not exist")
+            config_opts["BUILDTEST_MODULEPATH"] = tree_list
+
+    return config_opts
+
+
 def show_configuration():
     """This method display buildtest configuration to terminal and this
-    implements command buildtest show --config.
+       implements command buildtest show --config.
```

Comment:
might as well format this ``buildtest show --config`` in docstring so we dont forget

Reply:
Do you mean adding code blocks around it? Note that it's still there (the green line) it's just that there are spaces before it.

Reply:
what i mean is like https://github.com/HPC-buildtest/buildtest-framework/blob/master/buildtest/tools/config.py#L200  where command is surrounded by backtick so that the API can show that it is highlighted 

![image](https://user-images.githubusercontent.com/12942230/75115145-02929f00-562a-11ea-9fbd-9b8ee7423562.png)


https://buildtest.readthedocs.io/en/latest/api/buildtest.tools.config.html#buildtest.tools.config.func_config_edit

I know some docstrings are not formatted correctly and will need to fix it.